--- build/main.rs
+++ build/main.rs
@@ -130,2 +130,26 @@
 
 fn main() {
+    println!("cargo:rustc-cfg=openssl");
+    println!("cargo:rustc-cfg=ossl101");
+    println!("cargo:rustc-cfg=ossl102");
+    println!("cargo:rustc-cfg=ossl102f");
+    println!("cargo:rustc-cfg=ossl102h");
+    println!("cargo:rustc-cfg=ossl110");
+    println!("cargo:rustc-cfg=ossl110f");
+    println!("cargo:rustc-cfg=ossl110g");
+    println!("cargo:rustc-cfg=ossl110h");
+    println!("cargo:rustc-cfg=ossl111");
+    println!("cargo:rustc-cfg=ossl111b");
+    println!("cargo:rustc-cfg=ossl111c");
+    println!("cargo:rustc-cfg=ossl111d");
+    println!("cargo:rustc-cfg=ossl300");
+    println!("cargo:rustc-cfg=ossl310");
+    println!("cargo:rustc-cfg=ossl320");
+    println!("cargo:rustc-cfg=ossl330");
+    println!("cargo:rustc-cfg=ossl340");
+    println!("cargo:rustc-cfg=ossl350");
+    
+    println!("cargo:version_number=30500000"); // 3.5.0 (highest version openssl-sys 0.9.116 supports for OpenSSL 3.6.1 fallback)
+    println!("cargo:conf=");
+    return;
+
     let target = env::var("TARGET").unwrap();
